src/riak_kv_vnode.erl

```diff
@@ -840,6 +842,24 @@ handle_command({get_index_entries, Opts},
             {reply, ignore, State}
     end;
 
+%% For now, ignore async_put
+handle_command(?KV_W1C_BATCH_PUT_REQ{objs=Objs, type=Type},
+                From, State=#state{mod=Mod, idx=Idx, modstate=ModState}) ->
+    StartTS = os:timestamp(),
+    Context = {w1c_batch_put, From, Type, Objs, StartTS},
+    case Mod:batch_put(Context, Objs, [], ModState) of
+        {ok, UpModState} ->
+            lists:foreach(
+              fun({{Bucket, Key}, EncodedVal}) ->
+                      update_hashtree(Bucket, Key, EncodedVal, State),
```

Comment:
I'll double check thanks 

Sent from my iPhone

> On Mar 18, 2016, at 12:04 PM, Andrei Zavada <notifications@github.com> wrote:
> 
> In src/riak_kv_vnode.erl:
> 
> > @@ -840,6 +842,24 @@ handle_command({get_index_entries, Opts},
> >              {reply, ignore, State}
> >      end;
> >  
> > +%% For now, ignore async_put
> > +handle_command(?KV_W1C_BATCH_PUT_REQ{objs=Objs, type=Type},
> > +                From, State=#state{mod=Mod, idx=Idx, modstate=ModState}) ->
> > +    StartTS = os:timestamp(),
> > +    Context = {w1c_batch_put, From, Type, Objs, StartTS},
> > +    case Mod:batch_put(Context, Objs, [], ModState) of
> > +        {ok, UpModState} ->
> > +            lists:foreach(
> > +              fun({{Bucket, Key}, EncodedVal}) ->
> > +                      update_hashtree(Bucket, Key, EncodedVal, State),
> It's not safe to skip on intermediate updates and only call update_hashtree once, I assume?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>